syntax = "proto3";

package faster;

message RpcHeader
{
    bytes  service_name = 1;
    bytes  method_name  = 2;
    uint32 args_size    = 3;
}


message RpcResponseHeader 
{
    enum Type 
    {
        RESPONSE = 0;
        PUSHMSGS = 1;
    }

    Type   type         = 1;
    uint32 data_size    = 3;
}


// protoc rpcheader.proto --cpp_out=./